File: Java/src/objectiveta/objects/input/CandleStick.java
package objectiveta.objects.input;

public class CandleStick {

	private double open;
	private double high;
	private double low;
	private double close;
	
	public CandleStick(){}
	
	public CandleStick(double open, double high, double low, double close) 
	{
		this.open = open;
		this.high = high;
		this.low = low;
		this.close = close;
	}

	public double getOpen() {
		return open;
	}

	public void setOpen(double open) {
		this.open = open;
	}

	double getHigh() {
		return high;
	}

	void setHigh(double high) {
		this.high = high;
	}

	double getLow() {
		return low;
	}

	void setLow(double low) {
		this.low = low;
	}

	public double getClose() {
		return close;
	}

	public void setClose(double close) {
		this.close = close;
	}
	
}

File: cpp/Objects/Input/CandleStick.cpp
/*
 * CandleStick.cpp
 *
 *  Created on: 27 Aug 2018
 *      Author: William Ravenscroft
 *   Copyright: Objective-TA, 2018
 */

#include "../../../Objects/Input/CandleStick.h"

namespace objectiveta_objects_input {

CandleStick::CandleStick() {
    this->_open = 0.0;
	this->_high = 0.0;
	this->_low = 0.0;
	this->_close = 0.0;
}

CandleStick::CandleStick(double open, double high, double low, double close) {
	this->_open = open;
	this->_high = high;
	this->_low = low;
	this->_close = close;
}

double CandleStick::getOpen() {
	return _open;
}

double CandleStick::getHigh() {
	return _high;
}

double CandleStick::getLow() {
	return _low;
}

double CandleStick::getClose() {
	return _close;
}


CandleStick::~CandleStick() {
	delete _open;
	delete _high;
	delete _low;
	delete _close;
	delete this;
}

}


